svalbard/odata/src/main/java/org/n52/svalbard/odata/core/expr/Expr.java

```diff
@@ -187,6 +196,78 @@ default boolean isBoolean() {
         return Optional.empty();
     }
 
+    /**
+     * Check if this expression is a arithmetic expresion.
+     *
+     * @return if it is a arithmetic expression
+     */
+    default boolean isArithmetic() {
+        return false;
```

Comment:
These can delegate to `asXXX().isPresent()`. Then they don't have to be overwritten.

Reply:
refactored in d15ad2d